breadbox/breadbox/api/datasets.py

```diff
@@ -456,6 +459,53 @@ def get_dimensions(
     return search_index_entries
 
 
+@router.post(
+    "/dimension/data/",
+    operation_id="get_dimension_data",
+    response_model=DimensionDataResponse,
+)
+def get_dimension_data(
+    # The request body should be a SliceQuery with the following three fields:
+    dataset_id: Annotated[str, Body(description="The UUID or given ID of a dataset.")],
+    identifier: Annotated[
+        str,
+        Body(
+            description="A dimension identifier of the specified type (id, label, etc.)."
+        ),
+    ],
+    identifier_type: Annotated[
+        SliceQueryIdentifierType,
+        Body(
+            description="Denotes the type of identifier being used and the axis being queried."
+        ),
+    ],
+    db: SessionWithUser = Depends(get_db_with_user),
+    settings: Settings = Depends(get_settings),
+):
+    """
+    Load all values, IDs, and labels for a given dimension (specified by SliceQuery).
+    """
+    parsed_slice_query = SliceQuery(
+        dataset_id=dataset_id,
+        identifier=identifier,
+        identifier_type=identifier_type.name,
+    )
+    slice_values_by_id = slice_crud.get_slice_data(
+        db, settings.filestore_location, parsed_slice_query
+    )
+
+    # Load the labels separately, ensuring they're in the same order as the other values
+    slice_ids: list = slice_values_by_id.index.tolist()
+    labels_by_id = slice_crud.get_labels_for_slice_type(db, parsed_slice_query)
+    slice_labels = [labels_by_id[id] for id in slice_ids] if labels_by_id else None
```

Comment:
I always have a bias away from `Optional[X]` types such as `labels` seems to be. Thinking this through, I wonder under which conditions will `labels` be None?

I thought we had a design principal that all features have a user displayable name (using the id as a default if one wasn't explicitly stated)?

In which case this would be:

```
slice_labels = [labels_by_id[id] for id in slice_ids] if labels_by_id else slice_ids
```

Do you think it's important for this API make a distinction between IDs and labels? (As a consumer of the API, I'd typically want to have what's returned to me be as consistent as possible.)

Reply:
Good point! And yeah, you're right, that is the pattern we've been doing elsewhere. I'll update this to match

Reply:
Actually, as I was updating this, I realized that I don't currently have the behavior we want for column lookups (looking up column data is currently the case that's returning `None` for labels, and that's not what we want). 

When you load a column from a tabular dataset, that column should be indexed by a feature/sample given id. And that feature/sample will have corresponding labels. I'll work on updating that.

Reply:
One last follow-up on this: I realized there are a couple of open questions I have about tabular dataset indexes, and I would like to have a better understanding of how they work before I implement the crud functions we need to return labels for them here. 

I put a discussion topic in our breadbox meeting list: https://app.asana.com/0/1206925945720127/1208554346294686/f

And I put a task in Asana to follow up on this: https://app.asana.com/0/1165651979405609/1208554346294688/f

For now, I'll just do what you had originally suggested and return the IDs as labels for this particular case.